packages/components/progress/src/Progress.tsx

```diff
@@ -1,29 +1,57 @@
+import { useId } from '@radix-ui/react-id'
 import {
   Progress as ProgressPrimitive,
-  ProgressIndicator as ProgressPrimitiveIndicator,
   ProgressProps as ProgressPrimitiveProps,
 } from '@radix-ui/react-progress'
-import { ElementRef, forwardRef, PropsWithChildren } from 'react'
+import { cx } from 'class-variance-authority'
+import { ElementRef, forwardRef, PropsWithChildren, useMemo, useState } from 'react'
 
-export type ProgressProps = ProgressPrimitiveProps
+import { ProgressBar } from './ProgressBar'
+import { ProgressContext } from './ProgressContext'
+
+export interface ProgressProps extends ProgressPrimitiveProps {
+  isIndeterminate?: boolean
+}
 
 export const Progress = forwardRef<
   ElementRef<typeof ProgressPrimitive>,
   PropsWithChildren<ProgressProps>
->(({ value: valueProp, ...others }, ref) => {
-  const value = valueProp ?? 0
+>(
+  (
+    {
+      id: idProp,
+      className,
+      value: valueProp,
+      max = 100,
+      isIndeterminate = false,
+      children = <ProgressBar />,
+      ...others
+    },
+    ref
+  ) => {
+    const id = useId(idProp)
+    const [labelId, setLabelId] = useState<string>()
+
+    const value = useMemo(() => {
+      return { value: valueProp ?? 0, max, isIndeterminate, onLabelId: setLabelId }
+    }, [max, valueProp, isIndeterminate, setLabelId])
```

Comment:
I thought I already left a comment here, but apparently not 😅  
Do we really need to use `useMemo`  here? Doesn't seem like a heavy computation

Reply:
I normally add it to avoid possible re-renders (see [post](https://blog.agney.dev/useMemo-inside-context/)) but I guess that in this won't be that important

Reply:
Yes, I try to stay away from `useMemo`, `useCallback`, `useEffect` unless I really need them, as it tends to create noise imho, but it's not a blocker for me